src/Services/Message/Attachment.php

```diff
@@ -146,4 +156,19 @@ public function getData()
 
 		return $attachment->getData();
 	}
+
+	/**
+     * Returns attachment headers
+     * Contains Content-ID and X-Attachment-Id for embedded images
+     *
+     * @return array
+     */
+    public function getHeaderDetails($headers)
+    {
+        $headerDetails = [];
+        foreach ($headers as $header){
+            $headerDetails[$header->name] = $header->value;
```

Comment:
It looks good but requiring an object it's a little bit weird, I prefer using an array and doing key-value type of thing but I'm going to do it myself, you already identified the issue and created the solution and I appreciate that.